src/Kiota.Builder/Settings/VsCodeFileManagement.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Kiota.Builder.Settings;
+
+public static class VsCodeSettingsManager
+{
+    private static readonly JsonSerializerOptions options = new()
+    {
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+        WriteIndented = true,
+    };
+    private static readonly SettingsFileGenerationContext context = new(options);
+    public static async Task UpdateFileAsync(string fileUpdate, string fileUpdatePath, string fileUpdateKey, CancellationToken cancellationToken)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(fileUpdate);
+        ArgumentException.ThrowIfNullOrEmpty(fileUpdatePath);
+        ArgumentException.ThrowIfNullOrEmpty(fileUpdateKey);
+        Dictionary<string, object> settings = [];
+
+        // Read existing settings or create new if file doesn't exist
+        if (File.Exists(fileUpdatePath))
+        {
+            using var stream = File.OpenRead(fileUpdatePath);
+            settings = await JsonSerializer.DeserializeAsync(
+                stream,
+                context.DictionaryStringObject,
+                cancellationToken
+            ).ConfigureAwait(false) ?? [];
+        }
+
+        var fileUpdateDictionary = JsonSerializer.Deserialize(fileUpdate, context.DictionaryStringObject);
+        if (fileUpdateDictionary is not null)
+        {
+            if (fileUpdateDictionary.TryGetValue(fileUpdateKey, out var environmentVariables))
+            {
+                settings[fileUpdateKey] = environmentVariables;
+            }
+            else
+            {
```

Comment:
I suspect this is bug. Since the key here is `rest-client.environmentVariables`, and the json string that has been deserialized contains the `rest-client.environmentVariables` key(it was just serialized from a settings object), won't we have a json object serialized like this?

```json
{
    "rest-client.environmentVariables": 
        "rest-client.environmentVariables": 
            {
            }
}
```

Reply:
Not really. If the key already exists in the settings dictionary, the existing value will be overwritten. This is because the Dictionary's indexer (settings[fileUpdateKey]) will either add a new key-value pair if the key doesn't exist, or replace the value if the key does exist.

So there are two scenarios:
-If fileUpdateDictionary contains fileUpdateKey:The existing value in settings will be replaced with environmentVariables
-If fileUpdateDictionary doesn't contain fileUpdateKey:The existing value in settings will be replaced with the entire fileUpdateDictionary